science/exe.py

```diff
@@ -664,54 +477,35 @@ def build(
             f"requires at least 0.9.0."
         )
 
-    native_jump_path = (
-        a_scie.custom_jump(repo_path=use_jump)
-        if use_jump
-        else a_scie.jump(platform=platform_info.current)
-    )
-    with _temporary_directory(cleanup=not preserve_sandbox) as td:
-        for platform, lift_manifest in _export(
-            lift_config, application, dest_dir=td, platforms=platforms
-        ):
-            jump_path = (
-                a_scie.custom_jump(repo_path=use_jump)
-                if use_jump
-                else a_scie.jump(specification=application.scie_jump, platform=platform)
-            )
-            platform_export_dir = lift_manifest.parent
-            subprocess.run(
-                args=[str(native_jump_path), "-sj", str(jump_path), lift_manifest],
-                cwd=platform_export_dir,
-                stdout=subprocess.DEVNULL,
-                check=True,
-            )
-            src_binary_name = platform_info.current.binary_name(application.name)
-            dst_binary_name = (
-                platform.qualified_binary_name(application.name)
-                if use_platform_suffix
-                else platform.binary_name(application.name)
-            )
-            dest_dir.mkdir(parents=True, exist_ok=True)
-            dst_binary = dest_dir / dst_binary_name
-            shutil.move(src=platform_export_dir / src_binary_name, dst=dst_binary)
-            if hash_functions:
-                digests = tuple(
-                    hashlib.new(hash_function) for hash_function in sorted(set(hash_functions))
-                )
-                with dst_binary.open(mode="rb") as fp:
-                    for chunk in iter(lambda: fp.read(io.DEFAULT_BUFFER_SIZE), b""):
-                        for digest in digests:
-                            digest.update(chunk)
-                for digest in digests:
-                    dst_binary.with_name(f"{dst_binary.name}.{digest.name}").write_text(
-                        f"{digest.hexdigest()} *{dst_binary_name}{os.linesep}"
-                    )
-            click.echo(dst_binary)
+    with temporary_directory(cleanup=not preserve_sandbox) as td:
+        assembly_info = build.assemble_scies(
+            lift_config=lift_config,
+            application=application,
+            dest_dir=td,
+            platforms=platforms,
+            platform_info=platform_info,
+            use_jump=use_jump,
+            hash_functions=hash_functions,
+        )
+        dest_dir.mkdir(parents=True, exist_ok=True)
+
+        def move(path: Path) -> None:
+            dst = dest_dir / path.name
+            shutil.move(src=path, dst=dst)
+            click.echo(dst)
+
+        for scie_assembly in assembly_info.scies:
```

Comment:
This PR is a very nearly a pure move refactor. The only behavioral change is that the checksum file paths are now printed along with the scie binary path.